src/promptflow/promptflow/_sdk/_utils.py

```diff
@@ -787,14 +787,20 @@ def update_user_agent_from_env_var():
         OperationContext.get_instance().append_user_agent(os.environ["USER_AGENT"])
 
 
-def setup_user_agent_to_operation_context(user_agent):
-    """Setup user agent to OperationContext"""
+def setup_user_agent_to_operation_context(user_agent=None):
```

Comment:
Is this function redundant? Didn't we do more logic?
We can directly modify and use the `append_user_agent()`?

Reply:
If there must be a set_and_get, Is it more appropriate to place the get function in the OperationContext? The name has been changed to set_and_get_user_agent?

Reply:
This function will add ua from env var to context